common/src/main/java/com/groupbyinc/api/AbstractQuery.java

```diff
@@ -353,7 +353,7 @@ public Q addRefinementsByString(String refinementString) {
         if (refinementString == null) {
             return (Q) this;
         }
-        String[] filterStrings = refinementString.split("~");
+        String[] filterStrings = refinementString.split("~((?=[\\w]*[=:]))");
```

Comment:
Can you add a java comment here that describes the behaviour of the regex so that those who come later might understand?